help_to_heat/frontdoor/interface.py

```diff
@@ -108,23 +155,194 @@ def create_referral(self, session_id):
+        
+        paoNumbers = ""
+        if (paoStart != ""): paoNumbers += paoStart
+        if (paoEnd != ""): paoNumbers += ("–" + paoEnd);  # Use an en dash (–) for ranges
+        if (paoNumbers != ""): paoParts.append(paoNumbers)
+        
+        if (any(line_1_parts)):
+            line2Parts.extend(paoParts)
+            if ("STREET_DESCRIPTION" in lpi): line2Parts.append(lpi["STREET_DESCRIPTION"].title())
+        else:
+            line_1_parts.extend(paoParts)
+            if ("STREET_DESCRIPTION" in lpi): line_1_parts.append(lpi["STREET_DESCRIPTION"].title())
+
+        line1 = ", ".join(line for line in line_1_parts if line)
+        line2 = ", ".join(line for line in line2Parts if line)
+        
+        address = {
+            "address_line_1": line1,
+            "address_line_2": line2,
+            "town": lpi["TOWN_NAME"].title(),
+            "postcode": lpi["POSTCODE_LOCATOR"],
+            "local_custodian_code": lpi["LOCAL_CUSTODIAN_CODE"],
+            "uprn": lpi["UPRN"]
+        }
+
+        return address
+
+        
+    
+    def parse_dpa_to_address(self, dpa):
+        
+        line_1_parts = []
+        
+        if ("DEPARTMENT_NAME" in dpa): line_1_parts.append(dpa["DEPARTMENT_NAME"].title())
+        if ("ORGANISATION_NAME" in dpa): line_1_parts.append(dpa["ORGANISATION_NAME"].title())
+        if ("SUB_BUILDING_NAME" in dpa): line_1_parts.append(dpa["SUB_BUILDING_NAME"].title())
+        if ("BUILDING_NAME" in dpa): line_1_parts.append(dpa["BUILDING_NAME"].title())
+        if ("BUILDING_NUMBER" in dpa): line_1_parts.append(dpa["BUILDING_NUMBER"])
+
+        line2Parts = []
+        
+        if ("DOUBLE_DEPENDENT_LOCALITY" in dpa): line_1_parts.append(dpa["DOUBLE_DEPENDENT_LOCALITY"].title())
+        if ("DEPENDENT_LOCALITY" in dpa): line_1_parts.append(dpa["DEPENDENT_LOCALITY"].title())
+
+        if ("DEPENDENT_THOROUGHFARE_NAME" and "THOROUGHFARE_NAME" in dpa): line_1_parts.append(dpa["THOROUGHFARE_NAME"].title())
+        else:
+           line_1_parts.append(dpa["DEPENDENT_THOROUGHFARE_NAME"].title())
+           line2Parts.insert(0, dpa["THOROUGHFARE_NAME"].title())
+
+        line1 = ", ".join(line for line in line_1_parts if line)
+        line2 = ", ".join(line for line in line2Parts if line)
+
+        la_merges_dict = {       
+             "405": "440",
+             "410": "440", 
+             "415": "440", 
+             "425": "440",
+             "905": "940",
+             "915": "940",
+             "920": "940",
+             "910": "935",
+             "925": "935",
+             "930": "935",
+             "2705": "2745", 
+             "2710": "2745",
+             "2715": "2745",
+             "2720": "2745",
+             "2725": "2745",
+             "2730": "2745",
+             "2735": "2745",
+             "3305": "3300", 
+             "3310": "3300",
+             "3325": "3300",  
+             "3330": "3300", 
+             }
+        # Map:
```

Comment:
Is this copied from somewhere?If so, can we have a link to it in a comment? It's all a bit bewildering!

Reply:
Yes, it's hardcoded Local Authority mergers. The old codes are just mapped to the new codes.

Reply:
A comment is added to explain :)